File: web/_js/models/item.js
import {ValidatableBox} from "./validatableBox";

class Item extends ValidatableBox {
	constructor(length, width, height, quantity) {
		super(length, width, height);
		this.quantity = quantity;
	}

	get id() {
		return `${super.id}-${this.quantity}`;
	}

	getDimensions() {
		const dimensions = super.getDimensions();
		dimensions.push({name: 'Quantity', value: Number(this.quantity)});
		return dimensions;
	}

}

export default Item;

File: shared/src/Binacle.CompactNotation/Models/Item.cs
using System.Numerics;
using Binacle.Geometry;

namespace Binacle.CompactNotation;

// A concrete "LxWxH (X,Y,Z)" — dimensions and a position. What ParseItem/ParseItems return.
// Quantity is not a field: it is expanded into copies by ParseItems, so each Item is one placed thing.
public class Item<T> : IWithDimensions<T>, IWithCoordinates<T>
	where T : struct, IBinaryInteger<T>
{
	public T Length { get; set; }
	public T Width { get; set; }
	public T Height { get; set; }
	public T X { get; set; }
	public T Y { get; set; }
	public T Z { get; set; }
}
